response.go

```diff
@@ -2,11 +2,36 @@ package opensearchtools
 
 import "net/http"
 
+// RawOpenSearchResponse is a generic return type for an OpenSearch query which has meta fields common
+// to all response types as well as a generic Response field abstract across all response types.
+type RawOpenSearchResponse[T any] struct {
```

Comment:
Why do this wrapping? I'm not sure I see the value of this extra layer. To use a `OpenSearchResponse` I now have multiple levels to get to the actual response
```
resp.RawOpenSearchResponse.Response
```
It feels a little unneeded and clunky to me